schema-lending.graphql

```diff
@@ -570,19 +641,78 @@ type Market @entity {
   " Sum of all historical liquidations in USD "
   cumulativeLiquidateUSD: BigDecimal!
 
+  " Sum of all historical transfers in USD "
+  cumulativeTransferUSD: BigDecimal!
+
+  " Sum of all historical flashloans in USD "
+  cumulativeFlashloanUSD: BigDecimal!
+
+  " Total number of transactions "
+  transactionCount: Int!
+
+  " Total number of deposits "
+  depositCount: Int!
+
+  " Total number of withdrawals "
+  withdrawalCount: Int!
+
+  " Total number of borrows "
+  borrowCount: Int!
+
+  " Total number of repayments "
+  repayCount: Int!
+
+  " Total number of liquidations "
+  liquidationCount: Int!
+
+  " Total number of transfers "
+  transferCount: Int!
+
+  " Total number of flashloans "
+  flashloanCount: Int!
+
+  ##### Usage Data #####
+
+  " Number of cumulative unique users "
+  cumulativeUniqueUsers: Int!
+
+  " Number of cumulative depositors "
+  cumulativeUniqueDepositors: Int!
+
+  " Number of cumulative borrowers "
+  cumulativeUniqueBorrowers: Int!
+
+  " Number of cumulative liquidators (accounts that performed liquidation) "
+  cumulativeUniqueLiquidators: Int!
+
+  " Number of cumulative liquidatees (accounts that got liquidated) "
+  cumulativeUniqueLiquidatees: Int!
+
+  " Number of cumulative accounts that transferred positions (generally in the form of outputToken transfer) "
+  cumulativeUniqueTransferrers: Int!
+
+  " Number of cumulative accounts that performed flashloans "
+  cumulativeUniqueFlashloaners: Int!
+
   ##### Token Balances #####
 
-  " Amount of input tokens in the market "
-  inputTokenBalance: BigInt!
+  " Amount of input tokens in the market (same order as inputTokens) "
+  inputTokensBalance: [BigInt!]!
 
   " Price per share of input token in USD "
   inputTokenPriceUSD: BigDecimal!
 
-  " Total supply of output token "
-  outputTokenSupply: BigInt!
+  " Amount of input tokens borrowed in this market using variable interest rates (in native terms) "
+  variableBorrowedTokenBalance: BigInt
 
-  " Price per share of output token in USD "
-  outputTokenPriceUSD: BigDecimal!
+  " Amount of input tokens borrowed in this market using stable interest rates (in native terms) "
+  stableBorrowedTokenBalance: BigInt
+
+  " Total supply of output token (same order as outputTokens) "
+  outputTokensSupply: [BigInt!]
```

Comment:
I've had request before to change related fields into an array of structs instead of arrays of multiple fields, for example, something like:
```
type OutputToken @entity {
  supply: BigInt!
  priceUSD: BigDecimal!
  ...
}
```
This can eliminate the need to sort the items in the array to ensure alignment.

Reply:
Added `TokenBalances` to clean this up and allow for more custom relationships within markets

Reply:
See `TokenBalances` and let me know what you think

Reply:
Updated token balances to live as an array of structs in `market`. This essentially creates a mapping of inputToken:(multiple output tokens)

Reply:
using relation entity now instead, the `TokenData` seemed like a force